src/main/java/petproject/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package petproject.service.user;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import petproject.dto.user.CreateUserDto;
+import petproject.dto.user.UserDto;
+import petproject.exeption.RegistrationException;
+import petproject.mapper.UserMapper;
+import petproject.model.Role;
+import petproject.model.User;
+import petproject.repository.role.RoleRepository;
+import petproject.repository.user.UserRepository;
+
+@Service
```

Comment:
check other services and add `@Transactional`
```suggestion
@Service
@Transactional
```